### DISCUSSION TITLE
Lesson 7. Having troubles with FundMeTest.t.sol

### DISCUSSION THREAD

**Discussion by mibiot** [Cyfrin/foundry-full-course-cu#429] [Q&A] [answered]
Hi all,

I'm having troubles vs running FundMeTest.t.sol. If you take a look at 19-37 of the video, you ll see Patrick's code:
<img width="762" alt="Screenshot 2023-07-24 at 13 01 34" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/fed70bfe-620c-414e-ae42-5229a9e03f0c">

I just copy the code but getting instantly an error in the line 12:

<img width="492" alt="Screenshot 2023-07-24 at 13 14 23" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/e8a63d71-0114-4997-ba69-9f68d1445729">

when trying to compile it the error is as follows:

<img width="701" alt="Screenshot 2023-07-24 at 13 04 58" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/7706adc4-feb9-426a-8c93-4dc3e3ccb62b">

I have killed a few hours already trying to figure out what's wrong with my code since the code is identical to the Patric's version. FundMe file is taken from Lesson7's repository.

Please advice.

**Comment by shikhar229169** [comment, marked as answer]
It says that you need to pass an argument.  You made a constructor in your FundMe contract with a single argument, and you need to pass it while creating a new instance of contract.

Can you paste your FundMe contract here.

**Reply by mibiot** [reply]
// SPDX-License-Identifier: MIT
// 1. Pragma
pragma solidity ^0.8.18;
// 2. Imports

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
import {PriceConverter} from "./PriceConverter.sol";

// 3. Interfaces, Libraries, Contracts
error FundMe__NotOwner();

/**
 * @title A sample Funding Contract
 * @author Patrick Collins
 * @notice This contract is for creating a sample funding contract
 * @dev This implements price feeds as our library
 */
contract FundMe {
    // Type Declarations
    using PriceConverter for uint256;

    // State variables
    uint256 public constant MINIMUM_USD = 5 * 10 ** 18;
    address private immutable i_owner;
    address[] private s_funders;
    mapping(address => uint256) private s_addressToAmountFunded;
    AggregatorV3Interface private s_priceFeed;

    // Events (we have none!)

    // Modifiers
    modifier onlyOwner() {
        // require(msg.sender == i_owner);
        if (msg.sender != i_owner) revert FundMe__NotOwner();
        _;
    }

    // Functions Order:
    //// constructor
    //// receive
    //// fallback
    //// external
    //// public
    //// internal
    //// private
    //// view / pure

    constructor(address priceFeed) {
        s_priceFeed = AggregatorV3Interface(priceFeed);
        i_owner = msg.sender;
    }

    /// @notice Funds our contract based on the ETH/USD price
    function fund() public payable {
        require(
            msg.value.getConversionRate(s_priceFeed) >= MINIMUM_USD,
            "You need to spend more ETH!"
        );
        // require(PriceConverter.getConversionRate(msg.value) >= MINIMUM_USD, "You need to spend more ETH!");
        s_addressToAmountFunded[msg.sender] += msg.value;
        s_funders.push(msg.sender);
    }

    function withdraw() public onlyOwner {
        for (
            uint256 funderIndex = 0;
            funderIndex < s_funders.length;
            funderIndex++
        ) {
            address funder = s_funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // Transfer vs call vs Send
        // payable(msg.sender).transfer(address(this).balance);
        (bool success, ) = i_owner.call{value: address(this).balance}("");
        require(success);
    }

    function cheaperWithdraw() public onlyOwner {
        address[] memory funders = s_funders;
        // mappings can't be in memory, sorry!
        for (
            uint256 funderIndex = 0;
            funderIndex < funders.length;
            funderIndex++
        ) {
            address funder = funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // payable(msg.sender).transfer(address(this).balance);
        (bool success, ) = i_owner.call{value: address(this).balance}("");
        require(success);
    }

    /** Getter Functions */

    /**
     * @notice Gets the amount that an address has funded
     *  @param fundingAddress the address of the funder
     *  @return the amount funded
     */
    function getAddressToAmountFunded(
        address fundingAddress
    ) public view returns (uint256) {
        return s_addressToAmountFunded[fundingAddress];
    }

    function getVersion() public view returns (uint256) {
        return s_priceFeed.version();
    }

    function getFunder(uint256 index) public view returns (address) {
        return s_funders[index];
    }

    function getOwner() public view returns (address) {
        return i_owner;
    }

    function getPriceFeed() public view returns (AggregatorV3Interface) {
        return s_priceFeed;
    }
}

**Reply by mibiot** [reply]
I just copy-paste the file from https://github.com/Cyfrin/foundry-fund-me-f23/blob/main/src/FundMe.sol here, and the same code I use in VSC

**Reply by shikhar229169** [reply]
As you can see in the contract there is a constructor:
```solidity
constructor(address priceFeed) {
    s_priceFeed = AggregatorV3Interface(priceFeed);
    i_owner = msg.sender;
}
```

So while creating a new instance, you need to pass the priceFeed address.

**Reply by mibiot** [reply]
yeah!!! thx for helping! Got it! The issue was - I took the latest version of FundMe from repo, but in that moment Patric was using the old one from Remix.
